subsys/bluetooth/services/ots/ots_dir_list.c

```diff
@@ -22,8 +22,52 @@ LOG_MODULE_DECLARE(bt_ots, CONFIG_BT_OTS_LOG_LEVEL);
 
 static struct bt_ots_dir_list dir_lists[CONFIG_BT_OTS_MAX_INST_CNT];
 
-static void dir_list_object_encode(struct bt_gatt_ots_object *obj,
-				   struct net_buf_simple *net_buf)
+static size_t dir_list_object_record_size(const struct bt_gatt_ots_object *obj)
+{
+	uint16_t len;
+	size_t obj_name_len;
+
+	/* Record Length */
+	len = sizeof(len);
+
+	/* ID */
+	len += BT_OTS_OBJ_ID_SIZE;
+
+	/* Name length */
+	obj_name_len = strlen(obj->metadata.name);
+	__ASSERT(obj_name_len > 0 && obj_name_len <= CONFIG_BT_OTS_OBJ_MAX_NAME_LEN,
+		 "Dir list object len is incorrect %zu", obj_name_len);
+	len += sizeof(uint8_t);
```

Comment:
I reworked it a bit so it should be more obvious now. I rather keep it this way as a sanity check of the size. I think when/if we add or remove more elements from the record it will be easier to keep this function maintained versus the above shorthand.